tests/e2e/specs/shopper/filter-products-by-price.test.ts

```diff
@@ -0,0 +1,210 @@
+	await page.focus( selectors.frontend.priceMaxAmount );
+	await page.click( selectors.frontend.priceMaxAmount, {
+		clickCount: 3,
+	} );
+	await page.keyboard.type( '1.99' );
+	await page.$eval( selectors.frontend.priceMaxAmount, ( el ) =>
+		( el as HTMLElement ).blur()
+	);
+};
+
+describe( `${ block.name } Block`, () => {
+	describe( 'with All Products Block', () => {
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+
+			await insertBlock( block.name );
+			await insertBlock( 'All Products' );
+			await publishPost();
+
+			const link = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+			await page.goto( link );
+		} );
+
+		it( 'should render', async () => {
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await setMaxPrice();
+			await page.waitForTimeout( 1000 );
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( isRefreshed ).not.toBeCalled();
+			expect( products ).toHaveLength( 1 );
+			await expect( page ).toMatch( block.foundProduct );
+		} );
+	} );
+
+	describe( 'with PHP classic template ', () => {
+		const productCatalogTemplateId =
+			'woocommerce/woocommerce//archive-product';
+
+		useTheme( 'emptytheme' );
+		beforeAll( async () => {
+			await deleteAllTemplates( 'wp_template' );
+			await deleteAllTemplates( 'wp_template_part' );
+
+			await goToTemplateEditor( {
+				postId: productCatalogTemplateId,
+			} );
+			await insertBlock( block.name );
+			await saveTemplate();
+			await goToShopPage();
+		} );
+
+		afterAll( async () => {
+			await deleteAllTemplates( 'wp_template' );
+			await deleteAllTemplates( 'wp_template_part' );
+		} );
+
+		it( 'should render', async () => {
+			const products = await page.$$(
+				selectors.frontend.classicProductsList
+			);
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+
+			await page.waitForSelector( block.class + '.is-loading', {
+				hidden: true,
+			} );
+
+			await expect( page ).toMatch( block.foundProduct );
+			expect( isRefreshed ).not.toBeCalled();
+
+			await Promise.all( [
+				setMaxPrice(),
+				page.waitForNavigation( {
+					waitUntil: 'networkidle0',
+				} ),
+			] );
+
+			const products = await page.$$(
+				selectors.frontend.classicProductsList
+			);
+
+			const pageURL = page.url();
+			const parsedURL = new URL( pageURL );
+
+			expect( isRefreshed ).toBeCalledTimes( 1 );
+			expect( products ).toHaveLength( 1 );
```

Comment:
should we check here if the expected product is listed?

Reply:
[We check it 4 lines bottom](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6373/files/db0f316f85d061bc353142befce62f3378ea64c6#diff-008c918aa7487e2ab53be1c7032a692c9a904a019933eaef84634a0ff19f12fcR160)